syntax_and_semantics/operators.md

```diff
@@ -21,101 +45,271 @@ struct Vector2
+|---|---|---|---|
+| `*` | splat | `*foo` | no |
+| `**` | double splat | `**foo` | no |
+
+### Conditional
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `? :` | conditional | `a == b ? c : d` | no |
+
+### Assignments
+
+The assignment operator `=` assigns the value of the second operand to the first
+operand. The first operand is either a variable (in this case the operator can't
+be redefined) or a call (in this case the operator can be redefined).
+See [assignment](assignment.md) for details.
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `=` | variable assignment | `a = 1` | no |
+| `=` | call assignment | `a.b = 1` | yes |
+| `[]=` | index assignment | `a[0] = 1` | yes |
+
+### Combined Assignments
+
+The assignment operator `=` is the basis for all operators that combine an
+operator with assignment. The general form is `a <op>= b` and the compiler
+transform that into `a = a <op> b`.
+
+These transformations assume the receiver (`a`) is a variable. If it is a call,
+the replacements are semantically equivalent but the implementation is a bit more
+complex (introducing an anonymous temporary variable) and expects `a=` to be
+callable.
+
+The receiver can't be anything else than a variable or call.
+
+Exceptions are the logical operators:
+
+* `a ||= b` transforms to `a || (a = b)`
+* `a &&= b` transforms to `a && (a = b)`
+
+There is another special case when `a` is an index accessor (`[]`), it is
+changed to the nilable variant (`[]?` on the right hand side:
+
+* `a[i] ||= b` transforms to `a[i] = (a[i]? || b)`
+* `a[i] &&= b` transforms to `a[i] = (a[i]? && b)`
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `+=` | addition *and* assignment | `` | no |
+| `&+=` | wrapping addition *and* assignment | `` | no |
+| `-=` | subtraction *and* assignment | `` | no |
+| `&-=` | wrapping subtraction *and* assignment | `` | no |
+| `*=` | multiplication *and* assignment | `` | no |
+| `&*=` | wrapping multiplication *and* assignment | `` | no |
+| `/=` | division *and* assignment | `` | no |
+| `//=` | floor division *and* assignment | `` | no |
+| `%=` | modulo *and* assignment | `` | yes |
+| `\|=` | binary or *and* assignment | `` | no |
+| `&=` | binary and *and* assignment | `` | no |
+| `^=` | binary xor *and* assignment | `` | no |
+| `**=` | exponential *and* assignment | `` | no |
+| `<<=` | left shift *and* assignment | `` | no |
+| `>>=` | right shift *and* assignment | `` | no |
+| `\|\|=` | logical or *and*  assignment | `` | no |
+| `&&=` | logical and *and* assignment | `` | no |
 
-## Meaning
+### Index Accessors
 
-One can assign any meaning to the operators, but the convention is to follow the above ones to avoid cryptic code, or code that behaves in an unexpected way.
+Index accessors are used to query a value by index or key, for example an array
+item or map entry. The nilable variant `[]?` is supposed to return `nil` when
```

Comment:
Supposed? There are cases it won't be true?

Reply:
Depends entirely on the implementation.